pkg/sql/catalog/lease/name_cache.go

```diff
@@ -47,14 +48,15 @@ func (c *nameCache) get(
 	parentSchemaID descpb.ID,
 	name string,
 	timestamp hlc.Timestamp,
-) *descriptorVersionState {
+) (desc *descriptorVersionState, expiration hlc.Timestamp) {
 	c.mu.RLock()
```

Comment:
as described in As described in https://github.com/cockroachdb/cockroach/issues/109443#issuecomment-1694796474, `RWMutex` performs badly under high concurrency. Should we just change to a normal mutex?

Reply:
Let me try this out and see how it behaves without this change.

Reply:
No, it's similar to the old one: 
BenchmarkLeaseAcquireByNameCached-24    	 1624814	       734.4 ns/op

Also, there are no writes to the name cache in this scenario, so it should hit the fast path on the RWMutex (outside of system table activity).